### PR TITLE
Prepare queries for image uploading

### PR DESCRIPTION
**Pull request by gohabereg**

### CODE REVIEW

**Comment by neSpecc** [comment]
image

File: src/api/index.ts [hunk 1]
```diff
@@ -29,18 +29,56 @@ interface ApiCallSettings {
...
   { initial = false, force = false }: ApiCallSettings = {}
 ): Promise<any> {
-  const promise = axios.post(API_ENDPOINT, {
-    query: request,
-    variables
-  });
+  let promise: Promise<AxiosResponse>;
+
+  if (files && Object.keys(files).length) {
+    Object
```
Comment (neSpecc) [inline src/api/index.ts]: move this code to separated function

Comment (neSpecc) [reply src/api/index.ts]: (composing a FormData)

File: src/components/forms/ImageUploader.vue [hunk 2]
```diff
@@ -1,12 +1,51 @@
...
+  name: 'FormImageUploader',
+  mixins: [ uploadFile ],
+  props: {
+    image: {
+      type: String,
+      default: null
+    }
+  },
+  data: () => ({
+    imageSrc: null
```
Comment (neSpecc) [inline src/components/forms/ImageUploader.vue]: describe all props and data

File: src/mixins/uploadFile.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+export interface UploadFileOptions {
```
Comment (neSpecc) [inline src/mixins/uploadFile.ts]: add detailed description to this mixin

File: src/components/modals/WorkspaceCreationDialog.vue [hunk 4]
```diff
@@ -46,10 +47,14 @@ export default {
   },
   data() {
     return {
-      name: ''
+      name: '',
+      image: null,
     };
   },
   methods: {
+    onImageUpload(file) {
```
Comment (neSpecc) [inline src/components/modals/WorkspaceCreationDialog.vue]: jsdoc missed

File: src/components/account/Account.vue [hunk 5]
```diff
@@ -77,13 +82,19 @@ export default {
     user: state => state.user.data
   }),
   methods: {
+    onImageUpload(file) {
```
Comment (neSpecc) [inline src/components/account/Account.vue]: jsdocs are missed

File: src/components/account/Account.vue [hunk 6]
```diff
@@ -65,6 +68,8 @@ export default {
     return {
       name: user.name || '',
       email: user.email || '',
+      image: user.image || '',
```
Comment (neSpecc) [inline src/components/account/Account.vue]: jsdocs are missed

File: src/components/modals/WorkspaceCreationDialog.vue [hunk 7]
```diff
@@ -20,6 +20,7 @@
         />
         <ImageUploader
           class="workspace-creation-dialog__image-uploader"
+          @change="onImageUpload"
```
Comment (neSpecc) [inline src/components/modals/WorkspaceCreationDialog.vue]: better use `v-model`
https://vuejs.org/v2/guide/components.html#Using-v-model-on-Components

File: src/mixins/uploadFile.ts [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+      if (options.accept) {
+        input.setAttribute('accept', options.accept);
+      }
+
+      return new Promise(resolve => {
+        input.addEventListener('change', () => {
+          resolve(input.files);
+        });
+
+        input.click();
```
Comment (neSpecc) [inline src/mixins/uploadFile.ts]: I remember, that in some `safari` versions, that won't work without adding input to DOM. 

https://github.com/codex-team/ajax/pull/15

You can remove element after `onchange` fired: https://github.com/codex-team/ajax/blob/master/src/utils.js#L119

File: src/api/index.ts [hunk 9]
```diff
@@ -29,18 +30,28 @@ interface ApiCallSettings {
  * Makes request to API
  * @param {String} request - request to send
  * @param {Object} [variables] - request variables
+ * @param {Object} files - files to upload
```
Comment (nikmel2803) [inline src/api/index.ts]:
```suggestion
 * @param {Object} [files] - files to upload
```

File: src/api/projects/index.js [hunk 10]
```diff
@@ -7,7 +7,9 @@ import * as api from '../index.ts';
  * @return {Promise<Project.id>}
  */
 export async function createProject(projectInfo) {
-  return (await api.call(MUTATION_CREATE_PROJECT, projectInfo)).createProject;
+  const { image, ...rest } = projectInfo;
+
+  return (await api.call(MUTATION_CREATE_PROJECT, { ...rest }, { image })).createProject;
```
Comment (nikmel2803) [inline src/api/projects/index.js]:
```suggestion
  return (await api.call(MUTATION_CREATE_PROJECT, rest, { image })).createProject;
```

File: src/api/workspaces/index.ts [hunk 11]
```diff
@@ -29,15 +29,17 @@ interface CreateWorkspaceInput {
  * @returns {Promise<Workspace>} created workspace
  */
 export async function createWorkspace(workspaceInfo: CreateWorkspaceInput): Promise<Workspace> {
-  return (await api.call(MUTATION_CREATE_WORKSPACE, workspaceInfo)).createWorkspace;
+  const { image, ...rest } = workspaceInfo;
+
+  return (await api.call(MUTATION_CREATE_WORKSPACE, { ...rest }, { image })).createWorkspace;
```
Comment (nikmel2803) [inline src/api/workspaces/index.ts]:
```suggestion
  return (await api.call(MUTATION_CREATE_WORKSPACE, rest, { image })).createWorkspace;
```

**Pull request merged**